rest/client/rest.go

```diff
@@ -432,38 +394,33 @@ func (c *communicatorImpl) ExtendSpawnHostExpiration(ctx context.Context, hostID
 	return nil
 }
 
-// GetHosts gathers all active hosts and invokes a function on them
-func (c *communicatorImpl) GetHosts(ctx context.Context, f func([]*model.APIHost) error) error {
+// GetHosts gets all hosts matching filters
+func (c *communicatorImpl) GetHosts(ctx context.Context, data model.APIHostParams) ([]*model.APIHost, error) {
 	info := requestInfo{
 		method:  get,
-		path:    "hosts",
+		path:    "host/range",
 		version: apiVersion2,
 	}
 
-	p, err := newPaginatorHelper(&info, c)
+	resp, err := c.request(ctx, info, data)
 	if err != nil {
-		return err
+		err = errors.Wrapf(err, "error sending request to spawn host")
+		return nil, err
 	}
-
-	for p.hasMore() {
-		hosts := []*model.APIHost{}
-		resp, err := p.getNextPage(ctx)
-		if err != nil {
-			return err
-		}
-
-		err = util.ReadJSONInto(resp.Body, &hosts)
-		if err != nil {
-			return err
-		}
-
-		err = f(hosts)
-		if err != nil {
-			return err
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		errMsg := gimlet.ErrorResponse{}
+		if err := util.ReadJSONInto(resp.Body, &errMsg); err != nil {
+			return nil, errors.Wrap(err, "problem reading hosts error")
```

Comment:
Include the status code. (I can imagine getting an unreadable errMsg from the load balancer, for example, and knowing it was a 500 could be helpful in that case.)

Reply:
done